### PR TITLE
osd/scrub: Reintroduce scrub starts message

### PR DESCRIPTION
**Pull request by pdvian**
Whenever the scrubbing process starts for a PG, We used to log
scrub/deep-scrub "starts" message in the cluster log. We are
not seeing scrub "starts" messages anymore. Reintroduce scrub
starts message in order to calculate exact time taken to
scrub/deep-scrub the PG.

Fixes: https://tracker.ceph.com/issues/55798

Signed-off-by: Prashant D <pdhange@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ronen-fr** [comment]
@pdvian: only here, and not for 'after repair' scrubs?

File: src/osd/scrubber/pg_scrubber.cc [hunk 1]
```diff
@@ -191,6 +191,9 @@ void PgScrubber::initiate_regular_scrub(epoch_t epoch_queued)
     reset_epoch(epoch_queued);
     m_fsm->process_event(StartScrub{});
     dout(10) << "scrubber event --<< StartScrub" << dendl;
+    stringstream oss;
```
Comment (ronen-fr) [inline src/osd/scrubber/pg_scrubber.cc:194]: I would now prefer (as I did in all new code):
m_osds->clog->debug() << fmt::format(...

**Review by ronen-fr — CHANGES_REQUESTED** [review]
and see my other comment in the discussion

**Comment by pdvian** [comment]
> @pdvian: only here, and not for 'after repair' scrubs? A better location (that captures both) would be in set_scrub_begin_time()
Thanks Ronen. I did add it in set_scrub_begin_time() with __func__ logging earlier. Later from code realized that scrub message can be discarded e.g when PG changed state so in this case "starts" message will be misguiding. Is my understanding correct here ? Meaning when scrubbing starts we set begin time in set_scrub_begin_time but later scrubbing can be discarded later.

File: src/osd/scrubber/pg_scrubber.cc (hunk 1 above)
Comment (pdvian) [reply src/osd/scrubber/pg_scrubber.cc:194]: Sure. Let me change it in fmt::format.

**Comment by ronen-fr** [comment]
@pdvian: "Later from code realized that scrub message can be discarded e.g when PG state changed so in this case "starts" message will be misguiding."

This is doubly true for the current location you've chosen (which is earlier)...
Basically - you are never guaranteed that a scrub session will run to its end. There are many possible reasons for it to be scrubbed (sorry...). Even a benign occurrence, like multiple client writes to the specific range of objects being scrubbed, will cause it to abort.
As this was always so, having a 'scrub started' message without the scrub coming to conclusion is not an issue you should take into account.

**Comment by pdvian** [comment]
> @pdvian: "Later from code realized that scrub message can be discarded e.g when PG state changed so in this case "starts" message will be misguiding."
> 
> This is doubly true for the current location you've chosen (which is earlier)... Basically - you are never guaranteed that a scrub session will run to its end. There are many possible reasons for it to be scrubbed (sorry...). As this was always so, having a 'scrub started' message without the scrub coming to conclusion is not an issue you should take into account.

Thanks. Let me add it to set_scrub_begin_time.

**Review by ronen-fr — APPROVED** [review]
LGTM, apart from a clarification I asked for.

File: src/osd/scrubber/pg_scrubber.cc [hunk 2]
```diff
@@ -2145,6 +2145,10 @@ PgScrubber::PgScrubber(PG* pg)
 void PgScrubber::set_scrub_begin_time()
 {
   scrub_begin_stamp = ceph_clock_now();
+  m_osds->clog->debug() << fmt::format(
```
[the remark this answers is not in the document]
Comment (pdvian) [reply src/osd/scrubber/pg_scrubber.cc:2148]: Yes. I tested with downstream nautilus release.

**Pull request merged**